src/parser.rs

```diff
@@ -1128,7 +1115,8 @@ impl<'run, 'src> Parser<'run, 'src> {
     if attributes.is_empty() {
       Ok(None)
     } else {
-      Ok(Some((token.unwrap(), attributes.into_keys().collect())))
+      let attribute_set = AttributeSet::from_iter(attributes.into_keys());
```

Comment:
Can you do `collect` here instead of `from_iter`?

Reply:
Not quite sure what code change you want to see here.